internal/kafka/internal/services/kafka_tls_certificate_management/kafka_tls_certificate_management_service.go

```diff
@@ -0,0 +1,223 @@
+package kafka_tls_certificate_management
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
+	"github.com/caddyserver/certmagic"
+	"github.com/libdns/route53"
+)
+
+// CertificateManagementOutput is the output indicating the certificates references
+type CertificateManagementOutput struct {
+	TLSCertRef string
+	TLSKeyRef  string
+}
+
+// GetCertificateRequest is the certificate request object.
+type GetCertificateRequest struct {
+	TLSCertRef string
+	TLSKeyRef  string
+}
+
+// areCertRefsDefined returns true if all certificate references are defined.
+// otherwise returns false
+func (req GetCertificateRequest) areCertRefsDefined() bool {
+	return !(shared.StringEmpty(req.TLSCertRef) || shared.StringEmpty(req.TLSKeyRef))
+}
+
+// Certificate is the content of the certificate
+type Certificate struct {
+	TLSCert string
+	TLSKey  string
+}
+
+//go:generate moq -out kafka_tls_certificate_management_service_moq.go . KafkaTLSCertificateManagementService
+type KafkaTLSCertificateManagementService interface {
+	// ManageCertificate manages wildcard tls certificate of a given domain automatically.
+	// If the certificate does not exists, it generates them
+	// If they exists, checks if they've expired and in this case it renews them before they do.
+	// It returns the keys referencing the wildcard certificates location in the Storage
+	ManageCertificate(ctx context.Context, domain string) (CertificateManagementOutput, error)
+
+	// GetCertificate returns the tls certificate given the request.
+	// The certificate is returned from the underlying certificate storage when certificate management is automatic
+	// and that the certificate keys are defined i.e non empty string.
+	// Otherwise, the certificate is returned from the manual tls configuration files
+	GetCertificate(ctx context.Context, request GetCertificateRequest) (Certificate, error)
+
+	// RevokeCertificate revoke the certificate of given domain with a given reason
+	RevokeCertificate(ctx context.Context, domain string, reason CertificateRevocationReason) error
+
+	// IsKafkaExternalCertificateEnabled returns whether kafka external certificate is enabled
+	IsKafkaExternalCertificateEnabled() bool
+
+	// IsAutomaticCertificateManagementEnabled returns whether automatic certificate management is enabled
+	IsAutomaticCertificateManagementEnabled() bool
+}
+
+// certificateManagementClientWrapper wrapps the certmagic.Config struct so that we can easily unit test the
```

Comment:
What is certMagic? Can we get some information around it documented?

Reply:
The library choosen for auto cert generation - github.com/caddyserver/certmagic 
The library is in the import block - more documentation on it can be found on their github repo.

Reply:
Could we add some basic documentation around the type?

Reply:
Yes. Let me me see what can be added.

Reply:
I've added something, can you check?